addons/base/views.py

```diff
@@ -274,6 +275,20 @@ def get_auth(auth, **kwargs):
     if not provider_settings:
         raise HTTPError(httplib.BAD_REQUEST)
 
+    path = data.get('path')
+    version = data.get('version')
+
+    if path and auth.user:
+        mark_file_version_as_seen(auth.user, path, version)
+    if path and version:
```

Comment:
@Johnetordoff When viewing a file through the preprints app, `version` will be `None`, so we won't get in this block. This seems incorrect. Shouldn't we increment the analytics for the latest version in that case?

Reply:
Ideally the preprints app would just send the correct version, we can't just assume it's sending the latest version because we can download older versions via the app.

Reply:
If we download a specific version, then `version` will be set. The problem is when viewing a preprint through the preprints app. In that case, `version` isn't passed in the data.

Reply:
I thought that was why we were getting this info through the auth callback? So we could differentiate this stuff. If we don't have a version we probably need to have some kind of Pagecounter case where it's clear the verison is unknown and this doesn't disrupt the sum total of views for that file.

Reply:
@Johnetordoff We're using the auth callback because it always receives a render request, even when the requested file is cached.

`mark_file_version_as_seen` assumes that if `version` is `None`, the latest version is being requested. See https://github.com/CenterForOpenScience/osf.io/pull/8496/files/0390ac9d4659f89f614b4606130c3a11d5d52631#diff-8117cabdd0bd0d9cb074b789b7ce6dc9R373 .

Shouldn't the analytics code here do the same? Is there a situation where version is `None` and the latest version isn't being requested?

Reply:
@Johnetordoff Right, but is there any reason we shouldn't assume that the latest version is being requested when `version` is `None`, like `mark_file_version_as_seen` is doing?

Reply:
Ah I got you, yes you're correct if we are accurately getting a `None` as our verison it should be the latest.

Reply:
@Johnetordoff Can you send a PR to Erin's branch please?